test/ResearcherContract.test.js

```diff
@@ -127,4 +140,36 @@ contract("ResearcherContract", (accounts) => {
       assert.equal(researcherExists, false);
     });
   });
+
+  context("when add a research", () => {
+    it("should return error when is not a researcher", async () => {
+      await expectRevert(addResearch(), "Only allowed to researchers");
+    });
+
+    it("should add a work when is a researcher", async () => {
+      await addResearcher("Researcher A", resea1Address);
+      await addResearch(resea1Address);
+      const firstWork = await instance.worksCount();
+
+      assert.equal(firstWork, 1);
+    });
+
+    it("should add 1 to researcher publishedWorks", async () => {
+      await addResearcher("Researcher A", resea1Address);
+      await addResearch(resea1Address);
+      const researcher = await instance.getResearcher(resea1Address);
+
+      assert.equal(researcher.publishedWorks, 1);
+    });
+  });
+
+  describe("#getWorks", () => {
+    it("should return published works list", async () => {
+      await addResearcher("Researcher A", resea1Address);
```

Comment:
Linha 168 e 169 vc pode usar dentro de um `beforeEach`, assim vc deixa seu `it` mais limpo.

O `beforeEach` ficaria dentro do `describe`

Reply:
O problema é que aí teria que criar 2 describes diferentes

Reply:
```js
  describe("#getWorks", () => {
    beforeEach("", () => {
      await addResearcher("Researcher A", resea1Address);
      await addResearch(resea1Address);
    })

    it("should return published works list", async () => {
      const works = await instance.getResearches();

      assert.equal(works.length, 1);
    });
  });
```

Reply:
Mas isso não faria sentido fazer se tivesse mais testes rodando?? Por exemplo se tivesse mais 3 'its' com as mesmas variáveis, que aí reduziria a escrita para uma única vez. Nesse caso como só tem um 'it' não fica redundante?

Reply:
Pra legibilidade do `it` o ideal é que ele realize a ção que tem que realizar de forma direta.

Colocando o código abaixo dentro do `it` nós temos de ler essas linhas pra entender o que o it está testando.
```
      await addResearcher("Researcher A", resea1Address);
      await addResearch(resea1Address);
```
.

Colocando num before tbm nos tras mais flexibilidade pra quando ou se entrar mais its

Reply:
Lembrando que specs não custam `gas`. Não vão ser executados na `main net`. Quanto mais flexível  e limpo o spec melhor

Reply:
Alterado!!